src/abstractdataframe/show.jl

```diff
@@ -573,61 +573,193 @@ function _show(io::IO,
+            if nonmissingtype(types[i]) <: AbstractFloat
+                max_pad_i = 0
+                order_i   = zeros(Δr)
+                indices_i = zeros(Δr)
+
+                for k = 1:Δr
+                    # We need to process the top and bottom of the table because
+                    # we are cropping in the middle.
+
+                    if k ≤ Δr_lim
+                        kr = k
+                    else
+                        kr = num_rows - (k - Δr_lim) + 1
+                    end
+
+                    v = df[kr, i]
+
+                    order_v::Int = 0
+
+                    if v isa Number
+                        abs_v = abs(v)
+                        log_v::Int = (!isinf(v) && !isnan(v) && abs_v > 1) ? floor(Int, log10(abs_v)) : 0
+
+                        # If the order is higher than 5, then we print using
+                        # scientific notation.
+                        order_v = log_v > 5 ? 0 : floor(Int, log_v)
+
+                        # If the number is negative, we need to add an additional
+                        # padding to print the sign.
+                        v < 0 && (order_v += 1)
+                    end
+
+                    order_i[k]   = order_v
+                    indices_i[k] = kr
+
+                    order_v > max_pad_i && (max_pad_i = order_v)
+                end
+
+                push!(float_cols, i)
+                push!(indices, indices_i)
+                push!(padding, max_pad_i .- order_i)
+            end
+        end
+    end
+
+    # Create the formatter for floating point columns.
+    ft_float = (v, i, j)->_pretty_tables_float_formatter(v, i, j, float_cols,
+                                                         indices, padding,
+                                                         compact_printing)
+
+    # Make sure that `truncate` does not hide the type and the column name.
+    maximum_columns_width = [truncate == 0 ? 0 : max(truncate + 1, l, textwidth(t))
+                             for (l, t) in zip(names_len, types_str)]
+
+    # Check if the user wants to display a summary about the DataFrame that is
+    # being printed. This will be shown using the `title` option of
+    # `pretty_table`.
+    title = summary ? Base.summary(df) : ""
+
+    # If `rowid` is not `nothing`, then we are printing a data row. In this
+    # case, we will add this information using the row name column of
+    # PrettyTables.jl. Otherwise, we can just use the row number column.
+    if (rowid === nothing) || (ncol(df) == 0)
+        show_row_number = true
+        row_names = nothing
+    else
+        nrow(df) != 1 &&
+            throw(ArgumentError("rowid may be passed only with a single row data frame"))
+        show_row_number = false
+        row_names = [string(rowid)]
+    end
+
+    # Print the table with the selected options.
+    pretty_table(io, df, vcat(names_mat, types_str);
+                 alignment                   = :l,
```

Comment:
@nalimilan - here actually maybe it is better to keep alignment of `=`. What do you think?